### ISSUE TITLE
Issue install

### ISSUE THREAD

**Issue by Thomanphan** [mikeyy/nonoCAPTCHA#30]
#python3.6 basic.py 

python3.6 basic.py 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/sphinxbase/__init__.py", line 40, in <module>
    from .ad_pulse import *
  File "/usr/local/lib/python3.6/site-packages/sphinxbase/ad_pulse.py", line 38, in <module>
    _ad_pulse = swig_import_helper()
  File "/usr/local/lib/python3.6/site-packages/sphinxbase/ad_pulse.py", line 34, in swig_import_helper
    _mod = imp.load_module('_ad_pulse', fp, pathname, description)
  File "/usr/local/lib/python3.6/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/usr/local/lib/python3.6/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: /usr/local/lib/python3.6/site-packages/sphinxbase/_ad_pulse.cpython-36m-x86_64-linux-gnu.so: undefined symbol: PyCObject_FromVoidPtr

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "basic.py", line 4, in <module>
    from nonocaptcha.solver import Solver
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/solver.py", line 14, in <module>
    from nonocaptcha.audio import SolveAudio
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/audio.py", line 15, in <module>
    from nonocaptcha.speech import Amazon, Azure, Sphinx, DeepSpeech
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/speech.py", line 20, in <module>
    from pocketsphinx.pocketsphinx import Decoder
  File "/usr/local/lib/python3.6/site-packages/pocketsphinx/__init__.py", line 35, in <module>
    from sphinxbase import *
  File "/usr/local/lib/python3.6/site-packages/sphinxbase/__init__.py", line 42, in <module>
    from .ad_alsa import *
  File "/usr/local/lib/python3.6/site-packages/sphinxbase/ad_alsa.py", line 38, in <module>
    _ad_alsa = swig_import_helper()
  File "/usr/local/lib/python3.6/site-packages/sphinxbase/ad_alsa.py", line 34, in swig_import_helper
    _mod = imp.load_module('_ad_alsa', fp, pathname, description)
  File "/usr/local/lib/python3.6/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/usr/local/lib/python3.6/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: /usr/local/lib/python3.6/site-packages/sphinxbase/_ad_alsa.cpython-36m-x86_64-linux-gnu.so: undefined symbol: PyCObject_FromVoidPtr

**Comment by mikeyy** [comment]
Thanks for reporting your issue, 

Seeing that CObject isn't available since Python 3.2~, it's possible the SWIG version installed on your CentOS is outdated. 

Can you try installing the latest version of SWIG or at least >= 2.0.4? You may have to compile SWIG from source if your system's package manager only provides an older version.

After installing a more recent version of SWIG, you'll need to run "pip uninstall pocketsphinx" then rerun "pip install pocketsphinx" to use the newly installed SWIG.

Keep me updated.

**Comment by Thomanphan** [comment]
Issue fixed but I tried run examples and no result

```
# python3.6 test.py 
2018-09-13 10:00:41,301 0 Time elapsed: 0.1280350685119629
```
And no error display

**Comment by mikeyy** [comment]
This issue may be related to Chrome. Can you ensure you can run Chrome from the command line? 

The minimal code below should verify if you can launch Chrome, and possibly emit any errors on failure.

```python
import pyppeteer
import asyncio

launcher = pyppeteer.launcher.Launcher({'headless': False})
loop = asyncio.get_event_loop()
loop.run_until_complete(launcher.launch())
```

Hoping we can figure this out for you. Let me know if you need any extra guidance.

**Comment by Thomanphan** [comment]
# python3.6 check.py 
```
Traceback (most recent call last):
  File "check.py", line 8, in <module>
    loop.run_until_complete(launcher.launch())
  File "/usr/lib64/python3.6/asyncio/base_events.py", line 468, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.6/site-packages/pyppeteer/launcher.py", line 175, in launch
    self.browserWSEndpoint = self._get_ws_endpoint()
  File "/usr/local/lib/python3.6/site-packages/pyppeteer/launcher.py", line 195, in _get_ws_endpoint
    self.proc.stdout.read().decode()
pyppeteer.errors.BrowserError: Browser closed unexpectedly:
LaunchProcess: failed to execvp:
/root/.local/share/pyppeteer/local-chromium/571375/chrome-linux/nacl_helper
[9978:9978:0914/085417.126897:ERROR:nacl_fork_delegate_linux.cc(314)] Bad NaCl helper startup ack (0 bytes)

(chrome:9976): Gtk-WARNING **: cannot open display: 


```
I have change this line
`launcher = pyppeteer.launcher.Launcher({'headless': False, 'args': ['--no-sandbox','--user-data-dir', '--disable-setuid-sandbox']})`
if not 

# python3.6 check.py 
```
Traceback (most recent call last):
  File "check.py", line 8, in <module>
    loop.run_until_complete(launcher.launch())
  File "/usr/lib64/python3.6/asyncio/base_events.py", line 468, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.6/site-packages/pyppeteer/launcher.py", line 175, in launch
    self.browserWSEndpoint = self._get_ws_endpoint()
  File "/usr/local/lib/python3.6/site-packages/pyppeteer/launcher.py", line 195, in _get_ws_endpoint
    self.proc.stdout.read().decode()
pyppeteer.errors.BrowserError: Browser closed unexpectedly:
[10360:10360:0914/125953.994138:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.

But issue can't fix. 
# uname -a
`Linux test.domain.com 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

# rpm --query centos-release
`centos-release-7-5.1804.4.el7.centos.x86_64
`

**Comment by mikeyy** [comment]
Can you try running with headless set to True?

```python
launcher = pyppeteer.launcher.Launcher({'headless': True, 'args': 
    ['--no-sandbox',
    '--user-data-dir',
    '--disable-setuid-sandbox']
})
```

**Comment by Thomanphan** [comment]
Yes,But No display annything. no issue display too.

**Comment by mikeyy** [comment]
Let's see if we can output a webpages source, try this example:

```python
import pyppeteer
import asyncio

async def main():
    launcher = pyppeteer.launcher.Launcher({'headless': True, 'args': 
        ['--no-sandbox',
        '--user-data-dir',
        '--disable-setuid-sandbox']
    })
    browser = await launcher.launch()
    page = await browser.newPage()
    await page.goto("http://example.com/")
    print(await page.content())

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

**Comment by Thomanphan** [comment]
> Let's see if we can output a webpages source, try this example:
> 
> ```python
> import pyppeteer
> import asyncio
> 
> async def main():
>     launcher = pyppeteer.launcher.Launcher({'headless': True, 'args': 
>         ['--no-sandbox',
>         '--user-data-dir',
>         '--disable-setuid-sandbox']
>     })
>     browser = await launcher.launch()
>     page = await browser.newPage()
>     await page.goto("http://example.com/")
>     print(await page.content())
> 
> loop = asyncio.get_event_loop()
> loop.run_until_complete(main())
> ```

It's working normal

**Comment by mikeyy** [comment]
Well, at least that confirms there is a probably with my code somewhere. Will look for the problem and push a patch. Will keep you updated.

**Comment by Thomanphan** [comment]
but when try demo code still nothing display
`import asyncio
from nonocaptcha.solver import Solver

pageurl = "https://www.google.com/recaptcha/api2/demo"
sitekey = "6Le-wvkSAAAAAPBMRTvw0Q4Muexq9bi0DJwx_mJ-"

proxy = "185.222.134.52:3128"
auth_details = {
     "username": "",
     "password": ""
}
args = ["--timeout 5"]
options = {"ignoreHTTPSErrors": True,'headless': True, "args": ['--no-sandbox',
    '--user-data-dir',
    '--disable-setuid-sandbox']}
client = Solver(
     pageurl,
     sitekey,
     options=options,
     proxy=None,
     proxy_auth=None,
)

solution = asyncio.get_event_loop().run_until_complete(client.start())
if solution:
     print(solution)

`
result
```
# python3.6 test.py 
2018-09-14 14:15:48,088 0 Time elapsed: 0.1327223777770996
```

**Comment by mikeyy** [comment]
I've released 1.8.4, give it a few minutes before upgrading from pip. Otherwise upgrade by `pip install https://files.pythonhosted.org/packages/e0/8b/768857b9aa9b6c36ed523d04b9ec014e50b9f592aef2ff428b36a656560e/nonocaptcha-1.8.4-py3-none-any.whl`

This release doesn't fix your issue but will hopefully output errors that will inch us closer to determining the culprit and in fixing it.

**Comment by Thomanphan** [comment]
After update
```
#python3.6 test.py 
An error occurred during startup: 'NoneType' object has no attribute 'read'
2018-09-14 14:51:02,489 0 Time elapsed: 0.13216161727905273
```

**Comment by frosty00** [comment]
I actually had that same issue when I tried to run this on my debian server - I think it has something to do with pyppeteer. I think this problem can be solved by using docker, I opened an issue and will do some research into how to use docker myself.

**Comment by mikeyy** [comment]
> After update
> 
> ```
> #python3.6 test.py 
> An error occurred during startup: 'NoneType' object has no attribute 'read'
> 2018-09-14 14:51:02,489 0 Time elapsed: 0.13216161727905273
> ```

That's odd. I'm guessing this is occurring from this method on line 84 in `launcher.py`. Due to the vagueness of the error, I'm going to push another release that will provide us with more information using traceback. 

```python
    async def _get_ws_endpoint(self) -> str:
        url = self.url + '/json/version'
        while self.proc.returncode is None:
            await asyncio.sleep(0.1)
            try:
                with urlopen(url) as f:
                    data = json.loads(f.read().decode())
                break
            except URLError as e:
                continue
        else:
            raise BrowserError(
                'Browser closed unexpectedly:\n{}'.format(
                    self.proc.stdout.read().decode()
                )
            )
        return data['webSocketDebuggerUrl']
```

**Comment by mikeyy** [comment]
> I actually had that same issue when I tried to run this on my debian server - I think it has something to do with pyppeteer.
> 
> I think this problem can be solved by using docker; I opened an issue and will do some research into how to use docker myself.

Yeah, although it would be nice for newcomers to not have to go through the extra steps of setting up docker.

**Comment by mikeyy** [comment]
> After update
> 
> ```
> #python3.6 test.py 
> An error occurred during startup: 'NoneType' object has no attribute 'read'
> 2018-09-14 14:51:02,489 0 Time elapsed: 0.13216161727905273
> ```

I've released update 1.8.5 to out errors using traceback which should give us the full rundown on where this error is being produced.

**Comment by Thomanphan** [comment]
This is output
```
# python3.6 test.py 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/solver.py", line 53, in start
    self.browser = await self.get_new_browser()
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/solver.py", line 142, in get_new_browser
    browser = await self.launcher.launch()
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/launcher.py", line 72, in launch
    self.browserWSEndpoint = await self._get_ws_endpoint()
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/launcher.py", line 91, in _get_ws_endpoint
    self.proc.stdout.read().decode()
AttributeError: 'NoneType' object has no attribute 'read'
2018-09-15 00:29:29,843 0 Time elapsed: 0.13218045234680176
```

**Comment by mikeyy** [comment]
Odd, I will try removing that method and see if it begins to work for you then. I don't see it being any use for this library anyways, it was a new addition to the latest Pyppeteer update.

The update 1.8.6 has been released.

**Comment by Thomanphan** [comment]
I have update pyppeteer 0.0.24 issue same in 0.0.21 version
and output
```
# python3.6 test.py 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/solver.py", line 53, in start
    self.browser = await self.get_new_browser()
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/solver.py", line 142, in get_new_browser
    browser = await self.launcher.launch()
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/launcher.py", line 73, in launch
    self.browserWSEndpoint, self._loop, connectionDelay)
AttributeError: 'Launcher' object has no attribute 'browserWSEndpoint'
2018-09-15 01:52:41,795 0 Time elapsed: 0.02929973602294922
```

**Comment by mikeyy** [comment]
Well, that was an oversight on my part. I've released another update. Keep in mind this most likely won't fix your issue but may give us a reason on what's failing, considering the bug I'm trying to fix is due to BrowserError.

**Comment by mikeyy** [comment]
This is unfortunate. I'm grasping at clouds since I'm not able to reproduce the problem on my end. I will take a step back and consider the next actions to make for determining a more definite solution.

**Comment by frosty00** [comment]
@Thomanphan 

* Have you installed the latest chrome driver and is it in your $PATH env variable 

* What is the output of echo $DISPLAY? Are you able to view any other X commands? 

* Since pyppeteer is installed with pip if you are using it with ‘—user’ permissions you may need to add it to your path - ‘export PATH=$PATH:~/local/bin’

**Comment by mikeyy** [comment]
@Thomanphan Can you recheck your configuration file, nonocaptcha.yaml, and verify you have "headless: true"?

**Comment by Thomanphan** [comment]
@mikeyy wow, Working with this config
but what's result status ?
```
#python3.6 test.py 
2018-09-17 16:40:22,271 0 Starting solver with proxy None
2018-09-17 16:40:22,995 0 Clicking checkbox
2018-09-17 16:40:28,183 0 Clicking audio button
2018-09-17 16:40:28,631 0 Result: Detected
2018-09-17 16:40:28,642 0 Browser closed
2018-09-17 16:40:28,642 0 Time elapsed: 6.589920282363892
{'status': 'detected'}
```

**Comment by frosty00** [comment]
You need proxies. For now at least, I'm working on the image solving aspect.

**Comment by mikeyy** [comment]
> @mikeyy wow, Working with this config
> but what's result status ?
> 
> ```
> #python3.6 test.py 
> 2018-09-17 16:40:22,271 0 Starting solver with proxy None
> 2018-09-17 16:40:22,995 0 Clicking checkbox
> 2018-09-17 16:40:28,183 0 Clicking audio button
> 2018-09-17 16:40:28,631 0 Result: Detected
> 2018-09-17 16:40:28,642 0 Browser closed
> 2018-09-17 16:40:28,642 0 Time elapsed: 6.589920282363892
> {'status': 'detected'}
> ```

Sweet! Glad it's finally working now. It dawned on me that may be the issue. 👍 That just means your IP or the user-agent was marked as a bot. The best procedure is to use the library with proxies.

**Comment by mikeyy** [comment]
> You need proxies. For now at least, I'm working on the image solving aspect.

That too, image solving should alleviate the need for proxies for a longer duration. If not infinitesimally. Ultimately I think at some point, if too many failures arise from solving the image part incorrectly, it'll just reject any attempts of verifying - even if the images are correctly selected.

**Comment by mikeyy** [comment]
> I think this is first time to using this script on my pc , google can't block this

The library generates a new user-agent on the fly and sometimes that user-agent is marked as blocked instead of your IP. Usually, another attempt will proceed unblocked.

> Please check this code,is ok ? I tested this proxy it's working but has issue with this code

The code appears good to me. Can you verify the proxy you are using is able to access Google manually, without using the script?

**Comment by mikeyy** [comment]
>
```python
# python3.6 test.py 
2018-09-17 20:12:44,848 0 Starting solver with proxy http://179.43.132.61:80
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/base.py", line 61, in get_frames
    frame for frame in self.page.frames if "api2/anchor" in frame.url
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/solver.py", line 64, in start
    result = await self.solve()
  File "/usr/local/lib/python3.6/site-packages/nonocaptcha/solver.py", line 225, in solve
    await self.get_frames()
RuntimeError: coroutine raised StopIteration
2018-09-17 20:12:48,628 0 Browser closed
2018-09-17 20:12:48,628 0 Time elapsed: 3.9903957843780518```

By the way, I also get this error sometimes. Not sure why. I try to ensure each iframe is loaded in the document before proceeding but I guess sometimes there are false positives.

**Comment by Vivek-Patel06** [comment]
this error showing when i run code
also i tried solve that you are discuss this questions still i get this error
 
An error occurred during startup:
2020-04-02 04:54:45,183 0 Time elapsed: 0.021608829498291016
